tests/test_views.py

```diff
@@ -1,13 +1,85 @@
 from django.contrib.contenttypes.models import ContentType
 from django.contrib.sites.models import Site
+from django.test import RequestFactory
 
 from cms.models import Page, User
 from cms.test_utils.testcases import CMSTestCase
 from cms.utils.urlutils import admin_reverse
 
 from djangocms_navigation.constants import SELECT2_CONTENT_OBJECT_URL_NAME
-from djangocms_navigation.test_utils.factories import PageContentFactory
+from djangocms_navigation.test_utils.factories import (
+    MenuContentFactory,
+    PageContentFactory,
+    UserFactory,
+)
 from djangocms_navigation.test_utils.polls.models import Poll, PollContent
+from djangocms_navigation.views import MenuContentPreviewView
+
+
+class PreviewViewTestCases(CMSTestCase):
+    def setUp(self):
+        self.menu_content = MenuContentFactory()
+        self.preview_url = admin_reverse(
+            "djangocms_navigation_menuitem_preview",
+            kwargs={"menu_content_id": self.menu_content.id},
+        )
+
+    def test_view_url(self):
+        expcted_url = "/en/admin/djangocms_navigation/menuitem/{}/preview/".format(
+            self.menu_content.id
+        )
+        self.assertEqual(self.preview_url, expcted_url)
+
+    def test_view_anonymous_user(self):
+        response = self.client.get(self.preview_url)
+        expected_url = "/en/admin/login/?next=" + self.preview_url
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.url, expected_url)
+
+    def test_view_standard_user(self):
+        standard_user = self.get_standard_user()
+        with self.login_user_context(standard_user):
+            response = self.client.get(self.preview_url)
+            expected_url = "/en/admin/login/?next=" + self.preview_url
+            self.assertEqual(response.status_code, 302)
+            self.assertEqual(response.url, expected_url)
+
+    def test_view_super_user(self):
```

Comment:
I would suggest removing this test (if you follow the comments below, this test will not be needed) and perhaps dividing these tests into two TestCase classes. One which is just for testing permissions and the other for everything else to do with this view. This has two advantages. Firstly, you can put the creation of the superuser and log the user in in the `setUp` method in the "everything else TestCase" (which you can't in the permission test case since it's a different user in every test). Secondly, it allows more granular running of tests during development (sometimes it's useful to be able to run all tests in a smaller TestCase).